novawallet/Common/ViewController/CardLayoutPresentationController/Transitions/CardDismissingTransition.swift

```diff
@@ -0,0 +1,71 @@
+import UIKit
+import SoraUI
+
+final class CardDismissingTransition: NSObject {
+    private let transitionDuration: TimeInterval
+    private let animator: BlockViewAnimatorProtocol
+    private let dimmingViewTag: Int
+
+    init(
+        transitionDuration: TimeInterval = 0.25,
+        animator: BlockViewAnimatorProtocol,
+        dimmingViewTag: Int
+    ) {
+        self.transitionDuration = transitionDuration
+        self.animator = animator
+        self.dimmingViewTag = dimmingViewTag
+
+        super.init()
+    }
+}
+
+// MARK: UIViewControllerAnimatedTransitioning
+
+extension CardDismissingTransition: UIViewControllerAnimatedTransitioning {
+    func transitionDuration(using _: UIViewControllerContextTransitioning?) -> TimeInterval {
+        transitionDuration
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        guard
+            let sourceController = transitionContext.viewController(
+                forKey: UITransitionContextViewControllerKey.from
+            ),
+            let destinationController = transitionContext.viewController(
+                forKey: UITransitionContextViewControllerKey.to
+            )
+        else { return }
+
+        let coveredContextView: UIView
+
+        if let tabBarController = destinationController as? UITabBarController {
```

Comment:
It fills we are implementing that we are implementing custom presentation style not in the way Apple suggests to do it. This management of the view hierarchy must go in the presentation controller.